packages/admin-panel/src/table/DataFetchingTable/DataFetchingTable.jsx

```diff
@@ -0,0 +1,414 @@
+                {headers.map(
+                  (
+                    {
+                      getHeaderProps,
+                      render,
+                      isSorted,
+                      isSortedDesc,
+                      getSortByToggleProps,
+                      canSort,
+                      isButtonColumn,
+                    },
+                    i,
+                  ) => {
+                    return (
+                      <HeaderDisplayCell
+                        {...getHeaderProps(getSortByToggleProps())}
+                        // eslint-disable-next-line react/no-array-index-key
+                        key={`header-${i}`}
+                        isButtonColumn={isButtonColumn}
+                        width={visibleColumns[i].colWidth}
+                      >
+                        {render('Header')}
+                        {canSort && (
+                          <TableSortLabel
+                            active={isSorted}
+                            direction={isSortedDesc ? 'asc' : 'desc'}
+                            IconComponent={KeyboardArrowDown}
+                          />
+                        )}
+                      </HeaderDisplayCell>
+                    );
+                  },
+                )}
+              </TableRow>
+            ))}
+            <TableRow>
+              {displayFilterRow &&
+                visibleColumns.map(column => {
+                  return (
+                    <FilterCell
+                      key={column.id}
+                      column={column}
+                      onFilteredChange={onFilteredChange}
+                      filters={filters}
+                      width={column.colWidth}
+                    />
+                  );
+                })}
+            </TableRow>
+          </TableHead>
+          <TableBody {...getTableBodyProps()}>
+            {rows.map((row, index) => {
+              prepareRow(row);
+              return (
+                // eslint-disable-next-line react/no-array-index-key
+                <TableRow {...row.getRowProps()} key={`table-row-${index}`}>
+                  {row.cells.map(({ getCellProps, render }, i) => {
+                    return (
+                      <DisplayCell
+                        {...getCellProps()}
+                        // eslint-disable-next-line react/no-array-index-key
+                        key={`table-row-${index}-cell-${i}`}
+                        row={row}
+                        detailUrl={visibleColumns[i].isButtonColumn ? '' : detailUrl}
+                        getIsLink={getIsLink}
+                        width={visibleColumns[i].colWidth}
+                        getLink={getLink}
+                        isButtonColumn={visibleColumns[i].isButtonColumn}
+                      >
+                        {render('Cell')}
+                      </DisplayCell>
+                    );
+                  })}
+                </TableRow>
```

Comment:
I don't think this is strictly best practice - the cells are all clickable in this case which i think makes more sense semantically. And also some cells have buttons so we don't want the whole thing to be clickable because of this. I can however add a hover state to the row